### PR TITLE
Refactor cast by using convert function.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

File: util/types/convert.go [hunk 1]
```diff
@@ -97,19 +97,23 @@ func convertIntToInt(val, lowerBound, upperBound int64, tp byte) (converted int6
 	return
 }
 
+func convertUintToInt(val uint64, lowerBound, upperBound int64, tp byte) (converted int64, err error) {
+	if val > uint64(upperBound) {
+		converted = upperBound
+		err = overflow(val, tp)
+		return
```
Comment (zimulala) [inline util/types/convert.go]: return upperBound, overflow(val, tp) ?

Comment (qiuyesuifeng) [inline util/types/convert.go]: Yes, that seems clearer.
I only remove code from convertToInt function, there are some places have this problem, so i prefer to do it in next PR.

Comment (zimulala) [inline util/types/convert.go]: nice

**Comment by qiuyesuifeng** [comment]
PTAL

File: util/types/convert.go [hunk 2]
```diff
@@ -217,75 +253,22 @@ func typeError(v interface{}, target *FieldType) error {
...
+		mysql.TypeDate, mysql.TypeLonglong, mysql.TypeNewDecimal:
+		return true
 	}
+	return false
+}
+
+// Cast casts val to certain types and does not return error.
+func Cast(val interface{}, target *FieldType) (interface{}, error) {
+	if !isCastType(target.Tp) {
+		panic("unknown cast type, should never happen")
```
Comment (siddontang) [inline util/types/convert.go]: return error 
Rest LGTM

**Comment by shenli** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**